### ISSUE TITLE
Add Geppetto tutorial to existing HH Model tutorial 

### ISSUE THREAD

**Issue by joebowen** [openworm/hodgkin_huxley_tutorial#6]
Add tutorial to use existing HH model in Geppetto.

**Comment by pgleeson** [comment]
@joebowen, how about adding this project to [Open Source Brain](http://opensourcebrain.org/) and then anyone can view the details of the model with Geppetto (built in to OSB) without installing anything locally:

![selection_602](https://cloud.githubusercontent.com/assets/1556687/5814868/eced6c46-a087-11e4-90cb-fa4b2f0fece5.jpg)

Just sign up, add a new OSB project pointing to this github repo and then access the 3D view via the OSB 3D Explorer link. See [here](http://opensourcebrain.org/projects/acnet2?explorer=https%3A%2F%2Fraw.githubusercontent.com%2Fopenworm%2Fhodgkin_huxley_tutorial%2Fmaster%2FTutorial%2FSource%2FNML2_SingleCompHHCell.nml) for a preview of what you can expect.

By the way, I'd emphasize in the title/description of the project on OSB that it's a tutorial on how the HH model is described in NML2/Python as opposed to a tutorial on the background/history of the HH model itself (yet..?).

**Comment by pgleeson** [comment]
I can add this to OSB to allow access to the model in Geppetto. 

One philosophical point first though. This repo started out with the aim of comparing a HH model in Python to one in NML2 (by @joebowen). Now (with updates by @travs) it's aiming to be a general into/tutorial for HH. The former was useful, but the latter will be very useful, especially if it's well curated with links to OSB/Geppetto & latest nml tools (the Python/NML comparison can of course stay central to the tutorial). Is this the intention now? 

Relates to this issue: https://github.com/OpenSourceBrain/redmine/issues/116 and plans by me & @borismarin for a standalone HH tutorial on OSB.

**Comment by travs** [comment]
@pgleeson 

Yes, the aim of the tutorial has expanded beyond that first comparison to walk folks through the technical parts of the first 3 meetings of [the muscle neuron team](https://www.youtube.com/watch?v=6AhKE2Vg_Uw).

Some of the additions have been to generalize the tutorial with electrophysiology and math information, and this could certainly be augmented with background and history as you mention. 

The most recent additions also use PyNeuroML to produce plots describing the model (mostly used for I/V relationship now but this could also be expanded if needed).

I think adding OSB/Geppetto sections could only enhance the tutorial as well, though I'm not as familiar with those projects. It would be nice if someone using the tutorial could just hit "run" and see the simulation happen, which is one feature that I understand this would add.

**Comment by travs** [comment]
@joebowen Is this still something you would like to work on?

**Comment by joebowen** [comment]
@travs Yes, let me set aside some time this friday to spend working on this.

**Comment by travs** [comment]
@joebowen Ok excellent :-)

**Comment by pgleeson** [comment]
I've added this to OSB here: http://www.opensourcebrain.org/projects/hodgkin-huxley-tutorial. The 3D view in OSB Explorer (based on Geppetto) can be seen [here](http://www.opensourcebrain.org/projects/hodgkin-huxley-tutorial?explorer=https%3A%2F%2Fraw.githubusercontent.com%2Fopenworm%2Fhodgkin_huxley_tutorial%2Fmaster%2FTutorial%2FSource%2FNML2_SingleCompHHCell.nml). There would be more options in the window to investigate the cell info/conductance densities, etc. if the NeuroML network was split into *.net.nml, *.channel.nml, *.cell.nml files (like [here](https://github.com/OpenSourceBrain/GranuleCell/tree/master/neuroConstruct/generatedNeuroML2)), but this would probably involve updating the tutorial text too...

**Comment by joebowen** [comment]
Is there anything else that needs to be done on this issue?  Do we want to split the files up?

**Comment by travs** [comment]
@joebowen @pgleeson 
We should include a link in this tutorial somewhere that points to this new [OSB explorer](
http://www.opensourcebrain.org/projects/hodgkin-huxley-tutorial). Perhaps a new page?

Padraig, is there a way to embed the OSB explorer in the tutorial itself through an iframe or similar?

We can close this once we have at least that text/hyperlink inserted though.

**Comment by pgleeson** [comment]
Would be best to split the files first and then direct links could be added to OSB pages for the cell & channels. Also once simulation replay is enabled on OSB (by @adrianq and @tarelli) this will allow interactive replay of the model behaviour.

I can work on all this but it will be after [CNS](http://www.cnsorg.org/cns-2015-prague), end of July...

**Comment by pgleeson** [comment]
Info on interacting with/editing the model in Geppetto/[OSB explorer](http://www.opensourcebrain.org/projects/hodgkin-huxley-tutorial?explorer=https%3A%2F%2Fraw.githubusercontent.com%2Fopenworm%2Fhodgkin_huxley_tutorial%2Fmaster%2FTutorial%2FSource%2FHHCellNetwork.net.nml) version of the model can eventually be added in the [exercise description](https://github.com/openworm/hodgkin_huxley_tutorial/blob/master/Tutorial/_static/Exercises.rst)

Cc @tarelli @adrianq @borismarin

**Comment by pgleeson** [comment]
I've added the link to the OSB HH tutorial to the main README: https://github.com/openworm/hodgkin_huxley_tutorial/blob/master/README.md

Instead of a deep integration between the existing HH tutorial (on implementing HH in Python/NeuroML) I've pointed out that there are 2 related but independent tutorials in this repo.

The deep integration option would have required major changes in both tutorials... But more importantly they're for different audiences, the original one for those modellers wishing to work with Python/NeuroML, the OSB one for (potentially non computational) researchers/students who just want to understand the model and run some simulations without seeing/installing any code...

**Comment by pgleeson** [comment]
Hopefully the README now makes clear the different purposes/audiences of the 2 tutorials.
